### PR TITLE
100%

### PR DESCRIPTION
**Pull request by renatoi**
@src-code

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by renatoi** [comment]
2 ignored if/else statements because it is `hasOwnProperty()`:

https://github.com/gotwarlost/istanbul/blob/master/ignoring-code-for-coverage.md

File: src/lib/AtomicBuilder.js [hunk 1]
```diff
@@ -100,11 +100,11 @@ AtomicBuilder.prototype.run = function () {
...
         // check if we have some basic required keys in the atomic object
-        if (atomicObj.id.constructor !== String) {
-            throw new TypeError('Key `id` of atomic object must be a String. Object: ' + atomicObj);
+        if (!atomicObj.id || atomicObj.id.constructor !== String) {
+            throw new TypeError('Key `id` of atomic object is required and must be a String. Object: ' + atomicObj);
         }
-        if (atomicObj.type.constructor !== String) {
-            throw new TypeError('Key `type` of atomic object must be a String. Object: ' + atomicObj.id);
+        if (!atomicObj.type || atomicObj.type.constructor !== String) {
+            throw new TypeError('Key `type` of atomic object is required must be a String. Object: ' + atomicObj.id);
```
Comment (src-code) [inline src/lib/AtomicBuilder.js]: Missing the word "and" in there?

File: src/lib/AtomicBuilder.js [hunk 2]
```diff
@@ -156,9 +156,10 @@ AtomicBuilder.prototype.run = function () {
                     throw new TypeError('`Config ' + atomicObj.id + '.custom` must be an Array.');
                 }
                 currentConfigObj['custom-auto-suffix'].forEach(function (rule, index) {
-                    if (rule.constructor === Object) {
-                        rule.suffix = atomicObj.suffixType === 'numerical' ? index + 1 : String.fromCharCode(97 + index);
+                    if (rule.constructor !== Object) {
+                        throw new TypeError('`custo-auto-suffix` rule must be an Object.');
```
Comment (src-code) [inline src/lib/AtomicBuilder.js]: Typo in "custo-auto-suffix"...

**Comment by renatoi** [comment]
I had to do some modifications to the source code to handle if/else statements and also some bug fixes to previous tests.

**Comment by src-code** [comment]
:shipit:

**Pull request merged**